kedro-datasets/tests/spark/test_spark_dataset.py

```diff
@@ -440,6 +442,25 @@ def test_copy(self):
         assert spark_dataset_copy._file_format == "csv"
         assert spark_dataset_copy._save_args == {"mode": "overwrite"}
 
+    def test_dbfs_prefix_warning(self, monkeypatch, caplog):
```

Comment:
> Is [kedro-org/kedro#1801](https://github.com/kedro-org/kedro/issues/1801) an accurate description of what's happening?

This issue documents the error that users receive when the `exists_function` cannot resolve the given file path. There are evidently a few ways to trigger this, one of which is by leaving out the `/dbfs` prefix on Databricks. The user who posted this issue originally experienced the issue on Windows, though we don't have enough information to tell what caused it in that case.

The long-term solution to the general problem looks to be to refactor `SparkDataSet`. Specifically, we should bring the way that `exists_function` is determined in line with other datasets (https://github.com/kedro-org/kedro-plugins/issues/117). Doing so would resolve many cases in which this error is seen.

I created a new issue to link this PR to. I will leave the issue you mentioned open for now and see where we are after refactoring `SparkDataSet`.

> Why does it works when version is provided, is this also a permission issue about iterating the directory?

This could cause the same error in some cases, though in the case of deployment on Databricks this is unlikely to be the problem.

Reply:
You are right about this, though the linter complains that the file is long when these verbose tests are added (I originally had three separate tests). I would turn the warning off, but honestly I am a bit tired of ignoring the linter. There's not much point to using it if we ignore it every time it tells us off. My solution would be to add this slimmer method for now as this is quite urgent, then open an issue to split the test file. What do you think?

Reply:
Hmm, a better solution would be to split this back into three tests, add an ignore directive. Then split the file in a new issue and remove the ignore directive 🤔. This is what I will do.

Reply:
@jmholzer agree with this. Maybe we should just add more global ignores to the linter configuration for tests to make it less strict if you're finding that you continually need to add inline ignores.